tests/kernel/change_kernel.pm

```diff
@@ -0,0 +1,45 @@
+# SUSE's openQA tests
+#
+# Copyright © 2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+# Summary: Change the default kernel using a package
+# Maintainer: Richard Palethorpe <rpalethorpe@suse.com>
+use 5.018;
+use base "opensusebasetest";
+use testapi;
+use utils;
+use serial_terminal 'select_virtio_console';
+use qam;
+
+sub run {
+    my $self = shift;
+    my $repo = get_var('CHANGE_KERNEL_REPO');
+    my $pkg  = get_var('CHANGE_KERNEL_PKG') || 'kernel-default';
+
+    $self->wait_boot;
+    select_virtio_console();
+
+    # Avoid conflicts by removing any existing kernels
+    my @packages = kernel_packages;
+    zypper_call('-n rm ' . join(' ', @packages), exitcode => [0, 104], dumb_term => 1);
```

Comment:
currently you using kernel_packages function in three places. In all 3 places you need this function to get list of kernel packages to delete. For it looks more reasonable to replace this two lines with function `cleanup_kernel` which will contains code of kernel_packages and also line below which actually delete them